yt_dlp/extractor/tvw.py

```diff
@@ -0,0 +1,121 @@
+            'timestamp': 1704902400,
+            'upload_date': '20240110',
+            'location': 'Legislative Building',
+        }}, {
+        'url': 'https://tvw.org/video/ebeys-landing-state-park-2024081007/',
+        'md5': '71e87dae3deafd65d75ff3137b9a32fc',
+        'info_dict': {
+            'id': '2024081007',
+            'ext': 'mp4',
+            'title': 'Ebey\'s Landing State Park',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'description': 'md5:50c5bd73bde32fa6286a008dbc853386',
+            'timestamp': 1724310900,
+            'upload_date': '20240822',
+            'location': 'Ebey’s Landing State Park',
+        }},
+        {
+            'url': 'https://tvw.org/video/home-warranties-workgroup-2',
+            'info_dict': {
+                'id': '1999121000',
+                'ext': 'mp4',
+                'title': 'Home Warranties Workgroup',
+                'thumbnail': r're:^https?://.*\.jpg$',
+                'description': 'md5:861396cc523c9641d0dce690bc5c35f3',
+                'timestamp': 946389600,
+                'upload_date': '19991228',
+            },
+    }]
+
+    def _get_subtitles(self, response):
+        return {'en': [{'ext': 'vtt', 'url': response.get('captionPath')}]}
+
+    def _get_thumbnail(self, response):
+        return [{'url': response.get('videoThumbnail')}]
+
+    def _get_description(self, response):
+        return clean_html(response.get('description'))
+
+    def _get_js_code(self, video_id, webpage):
+        app_js_url = self._html_search_regex(
+            r'<script[^>]+src=[\"\'](?P<app_js>.+?)[\"\'][^>]+id=\"invintus-app-js\"',
+            webpage, 'app_js')
+        return self._download_webpage(app_js_url, video_id, 'Downloading app.js API key')
+
+    def _extract_formats(self, response, video_id, stream_url):
+        extract_formats = lambda url, video_id: self._extract_m3u8_formats(url, video_id, 'mp4')
+
+        try:
+            return extract_formats(stream_url, video_id)
+        except ExtractorError:
+            self.report_warning('Failed to parse the m3u8 stream. Falling back to the backup stream if it exists.')
+            try:
+                stream_url = traverse_obj(response, ('streamingURIs', 'backup'))
+                return extract_formats(stream_url, video_id)
+            except ExtractorError:
+                raise
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        app_js_code = self._get_js_code(video_id, webpage)
+
+        try:
+            api_key = self._search_regex(r'embedderAuth[\s]*:[\s]*{[^}]+vendorKey:\"(?P<token>\w+?)\"}',
+                                         app_js_code, 'token')
+        except RegexNotFoundError:
+            self.report_warning('Failed to extract the API key. Falling back to a hardcoded key.')
+            api_key = self.BACKUP_API_KEY
```

Comment:
The key has been the same [since 2018](https://web.archive.org/web/20181204140323/https://player.invintus.com/app.js). Lets hard code it for now and save the additional request. If they start regularly changing the key we can always add this back.